aiosmtpd/tests/test_starttls.py

```diff
@@ -0,0 +1,219 @@
+import asyncio
+import os
+import unittest
+from email.mime.text import MIMEText
+from smtplib import SMTP
+try:
+    import ssl
+    from asyncio import sslproto
+except ImportError:
+    _has_ssl = False
+else:
+    _has_ssl = True
+
+from .test_smtp import Controller as OldController, Sink
+
+from aiosmtpd.smtp import SMTP as SMTPProtocol
+
+
+class Controller(OldController):
+    def factory(self):
+        return SMTPProtocol(self.handler)
+
+
+class ReceivingHandler:
+    box = None
+
+    @asyncio.coroutine
+    def process_message(self, *args, **kws):
+        if not self.box:
+            self.box = []
+        self.box.append(args)
+
+
+def get_tls_context():
+    tls_context = ssl.create_default_context(ssl.Purpose.CLIENT_AUTH)
+    path = os.path.dirname(__file__)
+    path = os.path.join(path, 'certs')
+    tls_context.load_cert_chain(
```

Comment:
Please use `pkg_resources` to do this instead of `__file__`.  The former module makes accessing files relative to the import path immune to packaging problems.  E.g.

```
path = pkg_resources.resource_filename('aiosmtpd.tests.certs', 'server.crt')
```

It means you'll have to put an empty `__init__.py` in the `certs` directory to make it a subpackage, but that's fine.